target/linux/x86/patches-5.15/120-usb-net-for-fm350.patch

```diff
@@ -0,0 +1,59 @@
+Author: Ji Li <liji@huachen.link>
+Date:   Thu Jun 23 16:56:44 2022 +0800
+
+    USB: serial: option: add MEDIATEK FM350-GL module support
+    
+    Add usb product id of the MEDIATEK FM350-GL module.
+    
+    FM350-GL provides 2 mandatory interfaces to Linux host after enumeration.
+     - /dev/ttyUSB0: this is a serial interface for control path. User needs
+       to write AT commands to this device node to query status, set APN,
+       set PIN code, and enable/disable the data connection to 5G network.
+     - ethX: this is the data path provided as a RNDIS devices. After the
+       data connection has been established, Linux host can access 5G data
+       network via this interface.
+    
+    "RNDIS": RNDIS + ADB + AT (/dev/ttyUSB0) + MODEM COMs
+    
+    usb-devices output for 0x7127:
+    [ 2378.774113] usb 1-1: new high-speed USB device number 8 using xhci_hcd
+	[ 2378.923601] usb 1-1: New USB device found, idVendor=0e8d, idProduct=7127, bcdDevice= 0.01
+	[ 2378.923607] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
+	[ 2378.923611] usb 1-1: Product: FM350-GL
+	[ 2378.923614] usb 1-1: Manufacturer: Fibocom Wireless Inc.
+	[ 2378.936326] rndis_host 1-1:1.0 eth1: register 'rndis_host' at usb-0000:00:14.0-1, RNDIS device, 00:00:11:12:1
+3:14
+	[ 2378.938863] Discovery the interface for Fibocom & MEDIATEK.
+	…
+	[ 2378.940613] option 1-1:1.6: GSM modem (1-port) converter detected
+	[ 2378.940692] usb 1-1: GSM modem (1-port) converter now attached to ttyUSB0
+    
+    Signed-off-by: Ji Li <liji@huachen.link>
+---
+===================================================================
+--- linux-5.10.108.orig/drivers/usb/serial/option.c	2022-09-23 14:27:17.184479952 +0800
++++ linux-5.10.108/drivers/usb/serial/option.c	2022-09-23 14:33:22.907036605 +0800
+@@ -589,6 +589,7 @@
+ 
+ 
+ static const struct usb_device_id option_ids[] = {
++	{ USB_DEVICE(MEDIATEK_VENDOR_ID, 0x7127) },
```

Comment:
If you only want to support the interface number 6 I would prefer something like this:
````
{ USB_DEVICE_INTERFACE_NUMBER(MEDIATEK_VENDOR_ID, 0x7127, 0x6) },
````
See here for the matcher: https://elixir.bootlin.com/linux/v5.15.80/source/include/linux/usb.h#L1015